### PR TITLE
kubernetes-helmPlugins.helm-s3: 0.10.0 -> 0.14.0

### PR DESCRIPTION
**Pull request by Congee**
###### Description of changes

Added package version to the binary executable

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Congee** [comment]
Result of `nixpkgs-review pr 239838` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kubernetes-helmPlugins.helm-s3</li>
  </ul>
</details>

File: pkgs/applications/networking/cluster/helm/plugins/helm-s3.nix [hunk 1]
```diff
@@ -2,25 +2,25 @@
...
 
   # NOTE: Remove the install and upgrade hooks.
   postPatch = ''
     sed -i '/^hooks:/,+2 d' plugin.yaml
   '';
 
-  checkPhase = ''
-    make test-unit
-  '';
+  ldflags = [ "-s" "-w" "-X main.version=${version}" ];
```
Comment (yurrriq) [inline pkgs/applications/networking/cluster/helm/plugins/helm-s3.nix:21]: Nice one.

Comment (yurrriq) [inline pkgs/applications/networking/cluster/helm/plugins/helm-s3.nix:21]: Thanks for adding the version. Is there any particular reason to remove the `checkPhase`?

```suggestion
  checkPhase = ''
    make test-unit
  '';

  ldflags = [ "-s" "-w" "-X main.version=${version}" ];
```

Comment (Congee) [reply pkgs/applications/networking/cluster/helm/plugins/helm-s3.nix:21]: Thanks for reviewing. From my vague recollection, there was one test that did not pass. I wasn't sure how to solve it and the binary works find. I can close this PR if you deem the test necessary

Comment (yurrriq) [reply pkgs/applications/networking/cluster/helm/plugins/helm-s3.nix:21]: Thanks, I'll take a quick pass at it.

Comment (yurrriq) [reply pkgs/applications/networking/cluster/helm/plugins/helm-s3.nix:21]:
```
--- FAIL: TestDynamicBucketRegion (0.00s)
    --- FAIL: TestDynamicBucketRegion/existing_S3_bucket_URL_with_host_only_(no_key)_->_success (0.26s)
        session_test.go:58: 
                Error Trace:    /build/source/internal/awsutil/session_test.go:58
                Error:          Not equal: 
                                expected: "ap-southeast-2"
                                actual  : ""
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -ap-southeast-2
                                +
                Test:           TestDynamicBucketRegion/existing_S3_bucket_URL_with_host_only_(no_key)_->_success
    --- FAIL: TestDynamicBucketRegion/existing_S3_bucket_URL_with_key_->_success (0.32s)
        session_test.go:58: 
                Error Trace:    /build/source/internal/awsutil/session_test.go:58
                Error:          Not equal: 
                                expected: "ap-southeast-2"
                                actual  : ""
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -ap-southeast-2
                                +
                Test:           TestDynamicBucketRegion/existing_S3_bucket_URL_with_key_->_success
```

Comment (yurrriq) [reply pkgs/applications/networking/cluster/helm/plugins/helm-s3.nix:21]: Here's may hackish approach...

```suggestion
  # NOTE: make test-unit, but skip awsutil, which needs internet access
  checkPhase = ''
    go test $(go list ./... | grep -vE '(awsutil|e2e)')
  '';

  ldflags = [ "-s" "-w" "-X main.version=${version}" ];
```

Comment (Congee) [reply pkgs/applications/networking/cluster/helm/plugins/helm-s3.nix:26]: thanks

**Review by yurrriq — APPROVED** [review]
Thanks for the update and for your patience!

**Comment by jaudiger** [comment]
I installed those two packages on a nix-darwin based system relying on unstable channel: "kubernetes-helm" and "kubernetes-helmPlugins.helm-s3".

But I'm not sure how I could take advantage to the helm-s3 plugin package. The following command returned:

```bash
> help plugin list
NAME	VERSION	DESCRIPTION
```

But I was expecting to see the helm-s3 plugin list here, did I mis-interpret something ?

**Pull request merged**